ceph.yaml

```diff
@@ -0,0 +1,212 @@
+      recurse-submodules: true
+
+  - uses: patch
+    with:
+      patches: fix-submodule.patch
+
+  - uses: cmake/configure
+    with:
+      opts: |
+        -Wno-dev \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=/usr \
+        -DCMAKE_CXX_STANDARD=17 \
+        -DCMAKE_INSTALL_LOCALSTATEDIR=/var \
+        -DPython3_LIBRARIES=/usr/lib/python3.11 \
+        -DCMAKE_INSTALL_SYSCONFDIR=/etc \
+        -DNINJA_MAX_COMPILE_JOBS=$JOBS \
+        -DALLOCATOR=libc \
+        -DWITH_GTEST_PARALLEL=ON \
+        -DWITH_PYTHON3=3.11 \
+        -DENABLE_GIT_VERSION=ON \
+        -DLUA_LIBRARIES=/usr/lib/lua5.4/liblua.so \
+        -DWITH_BABELTRACE=OFF \
+        -DWITH_JAEGER=OFF \
+        -DWITH_LTTNG=OFF \
+        -DWITH_MANPAGE=ON \
+        -DWITH_RADOSGW_AMQP_ENDPOINT=OFF \
+        -DWITH_RADOSGW_SELECT_PARQUET=ON \
+        -DWITH_RADOSGW_POSIX=OFF \
+        -DWITH_RDMA=OFF \
+        -DWITH_REENTRANT_STRSIGNAL=ON \
+        -DWITH_SEASTAR=OFF \
+        -DWITH_SPDK=OFF \
+        -DWITH_SYSTEMD=ON \
+        -DWITH_CEPHFS_SHELL=ON \
+        -DWITH_SYSTEM_ARROW=ON \
+        -DWITH_SYSTEM_BOOST=OFF \
+        -DWITH_SYSTEM_FMT=OFF \
+        -DWITH_SYSTEM_LIBURING=ON \
+        -DWITH_SYSTEM_NPM=ON \
+        -DWITH_MGR_DASHBOARD_FRONTEND=ON \
+        -DWITH_SYSTEM_ROCKSDB=OFF \
+        -DWITH_SYSTEM_ZSTD=ON \
+        -DWITH_QATLIB=OFF \
+        -DWITH_QATZIP=OFF \
+        -DWITH_TESTS="$(want_check && echo ON || echo OFF)" \
+        -DWITH_THREAD_SAFE_RES_QUERY=ON
+
+  - runs: cmake --build output --target common-options-objs
+
+  - uses: cmake/build
+
+  - uses: cmake/install
+
+  - uses: strip
+
+subpackages:
+  - name: ${{package.name}}-dev
+    dependencies:
+      runtime:
+        - ${{package.name}}
+    pipeline:
+      - uses: split/dev
+
+  - name: ${{package.name}}-test-doc
+    pipeline:
+      - uses: split/manpages
+    description: ceph manpages
+    dependencies:
+      provides:
+        - ceph-doc=${{package.full-version}}
+    test:
+      pipeline:
+        - uses: test/docs
+
+test:
+  environment:
+    contents:
+      packages:
+        - sudo-rs
+        - py3.11-packaging
+        - py3.11-pyyaml
+        - systemd
+        - docker
+        - lvm2-dev
+  pipeline:
+    - name: Ensure version build
+      runs: |
+        ceph --version  | grep ${{package.version}}
+        cephadm version | grep ${{package.version}}
+        ceph-fuse --version | grep ${{package.version}}
+        ceph-mds --version  | grep ${{package.version}}
+        ceph-mgr --version | grep ${{package.version}}
+        ceph-mon --version | grep ${{package.version}}
+        ceph-osd --version | grep ${{package.version}}
+        ceph-objectstore-tool --version | grep ${{package.version}}
+        radosgw --version | grep ${{package.version}}
+        radosgw-admin --version | grep ${{package.version}}
+        rbd --version | grep ${{package.version}}
+    - uses: test/kwok/cluster
```

Comment:
[regress-stack ](https://github.com/canonical/regress-stack/blob/main/src/regress_stack/modules/ceph.py)(used in Ubuntu) has a module for deploying ceph in a single node deployment - its python but should be translatable into something for a test in wolfi